### PR TITLE
PA-2822 Populated Building LeasedLandMetadata

### PR DESCRIPTION
**Pull request by Fosol**
Existing buildings within PIMS would have issues when viewing their associated land.  This update resolves this.

Added new DB migration `v01.08.03`

image

### CODE REVIEW

File: backend/api/Areas/Property/Mapping/Building/BuildingMap.cs [hunk 1]
```diff
@@ -65,14 +65,14 @@ public void Register(TypeAdapterConfig config)
...
+                    if (src.LeasedLandMetadata == null)
                     {
                         dest.LeasedLandMetadata = new List<Model.LeasedLandMetadataModel>();
                         return;
                     }
-                    var metadata = JsonSerializer.Deserialize<IEnumerable<Entity.Models.LeasedLandMetadata>>(src.LeasedLandMetadata, _serializerOptions);
+                    var metadata = JsonSerializer.Deserialize<IEnumerable<Entity.Models.LeasedLandMetadataModel>>(src.LeasedLandMetadata, _serializerOptions);
 
-                    dest.LeasedLandMetadata = metadata.Where(m => m != null).Select(l => new Model.LeasedLandMetadataModel { OwnershipNote = l.OwnershipNote, ParcelId = l.ParcelId, Type = l.Type });
+                    dest.LeasedLandMetadata = metadata.Where(m => m != null).Select(l => new Model.LeasedLandMetadataModel { OwnershipNote = l.OwnershipNote, ParcelId = l.ParcelId, Type = (int)l.Type });
```
Comment (Fosol) [inline backend/api/Areas/Property/Mapping/Building/BuildingMap.cs:75]: Needs to be converted to an `int` for the frontend code.  Most of our `enum` values are converted to their string names, but here they are not.

File: backend/api/Areas/Property/Models/Building/LeasedLandMetadataModel.cs [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
         /// <summary>
         /// get/set - Note related to the ownership of the land associated to this building.
         /// </summary>
         public string OwnershipNote { get; set; }
+
         /// <summary>
         /// get/set - An id used to associate this metadata with a parcel.
         /// </summary>
-        public string ParcelId { get; set; }
+        public int ParcelId { get; set; }
```
Comment (Fosol) [inline backend/api/Areas/Property/Models/Building/LeasedLandMetadataModel.cs:17]: Correct type.

File: backend/entities/LeasedLandTypes.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+namespace Pims.Dal.Entities
+{
+    /// <summary>
+    /// LeasedLandTypes enum, provides the valid building leased land types that can be used.
+    /// </summary>
+    public enum LeasedLandTypes
```
Comment (Fosol) [inline backend/entities/LeasedLandTypes.cs:6]: Replacing `int` with compile time safe enum

File: backend/entities/Models/LeasedLandMetadataModel.cs [hunk 4]
```diff
@@ -1,22 +1,25 @@
...
         /// get/set - An id used to associate this metadata with a parcel.
         /// </summary>
-        public string ParcelId { get; set; }
+        public int ParcelId { get; set; }
+
         /// <summary>
         /// get/set - the lease type
         /// </summary>
-        public int Type { get; set; }
+        public LeasedLandTypes Type { get; set; }
```
Comment (Fosol) [inline backend/entities/Models/LeasedLandMetadataModel.cs:22]: Ensures the correct values are used.

File: frontend/src/actions/parcelsActions.ts [hunk 5]
```diff
@@ -62,7 +62,7 @@ export interface IAddress {
   postal: string;
 }
 
-export enum LeasedLand {
+export enum LeasedLandTypes {
```
Comment (Fosol) [inline frontend/src/actions/parcelsActions.ts:65]: Renamed for consistency

**Review by devinleighsmith — APPROVED** [review]

**Review by goemen — APPROVED** [review]
lgtm

File: backend/entities/LeasedLandTypes.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    {
+        /// <summary>
+        /// The ministry owns the land the building is on.
+        /// </summary>
+        Owned = 0,
+        /// <summary>
+        /// The ministry leases th eland the building is on.
```
Comment (agalbrai) [inline backend/entities/LeasedLandTypes.cs:13]: nitpicky but "th eland"

File: backend/entities/LeasedLandTypes.cs [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+        /// </summary>
+        Leased = 1,
+        /// <summary>
+        /// The relationship with the land is complex.
```
Comment (agalbrai) [inline backend/entities/LeasedLandTypes.cs:17]: "it's complicated" haha

**Review by jdtoombs — APPROVED** [review]

**Pull request merged**